interfaces/builtin/storage_framework_service_test.go

```diff
@@ -0,0 +1,108 @@
+version: 1.0
+apps:
+ app:
+  command: foo
+  slots: [storage-framework-service]
+`
+	// a storage-framework-service slot on the core snap (as automatically added on classic)
+	const storageFrameworkServiceMockClassicSlotSnapInfoYaml = `name: core
+type: os
+slots:
+ storage-framework-service:
+  interface: storage-framework-service
+`
+	const mockPlugSnapInfo = `name: client
+version: 1.0
+apps:
+ app:
+  command: foo
+  plugs: [storage-framework-service]
+`
+	s.iface = &builtin.StorageFrameworkServiceInterface{}
+	// storage-framework-service snap with storage-framework-service slot on an core/all-snap install.
+	snapInfo := snaptest.MockInfo(c, storageFrameworkServiceMockCoreSlotSnapInfoYaml, nil)
+	s.coreSlot = &interfaces.Slot{SlotInfo: snapInfo.Slots["storage-framework-service"]}
+	// storage-framework-service slot on a core snap in a classic install.
+	snapInfo = snaptest.MockInfo(c, storageFrameworkServiceMockClassicSlotSnapInfoYaml, nil)
+	s.classicSlot = &interfaces.Slot{SlotInfo: snapInfo.Slots["storage-framework-service"]}
+
+	plugSnap := snaptest.MockInfo(c, mockPlugSnapInfo, nil)
+	s.plug = &interfaces.Plug{PlugInfo: plugSnap.Plugs["storage-framework-service"]}
+}
+
+func (s *StorageFrameworkServiceInterfaceSuite) TestName(c *C) {
+	c.Check(s.iface.Name(), Equals, "storage-framework-service")
+}
+
+func (s *StorageFrameworkServiceInterfaceSuite) TestSanitizeIncorrectInterface(c *C) {
+	c.Check(func() { s.iface.SanitizeSlot(&interfaces.Slot{SlotInfo: &snap.SlotInfo{Interface: "other"}}) },
+		PanicMatches, `slot is not of interface "storage-framework-service"`)
+	c.Check(func() { s.iface.SanitizePlug(&interfaces.Plug{PlugInfo: &snap.PlugInfo{Interface: "other"}}) },
+		PanicMatches, `plug is not of interface "storage-framework-service"`)
+}
+
+func (s *StorageFrameworkServiceInterfaceSuite) TestUsedSecuritySystems(c *C) {
+	// connected slots have a non-nil security snippet for apparmor
+	apparmorSpec := &apparmor.Specification{}
+	err := apparmorSpec.AddConnectedSlot(s.iface, s.plug, s.coreSlot)
+	c.Assert(err, IsNil)
+	c.Assert(apparmorSpec.SecurityTags(), DeepEquals, []string{"snap.storage-framework-service.app"})
+	c.Assert(apparmorSpec.SnippetForTag("snap.storage-framework-service.app"), testutil.Contains, `interface=com.canonical.StorageFramework`)
+
+	// slots have no permanent snippet on classic
+	apparmorSpec = &apparmor.Specification{}
+	err = apparmorSpec.AddConnectedSlot(s.iface, s.plug, s.classicSlot)
+	c.Assert(err, IsNil)
+	c.Assert(apparmorSpec.SecurityTags(), HasLen, 0)
+
+	// slots have a permanent non-nil security snippet for apparmor
+	apparmorSpec = &apparmor.Specification{}
+	err = apparmorSpec.AddPermanentSlot(s.iface, s.coreSlot)
+	c.Assert(err, IsNil)
+	c.Assert(apparmorSpec.SecurityTags(), DeepEquals, []string{"snap.storage-framework-service.app"})
+	c.Assert(apparmorSpec.SnippetForTag("snap.storage-framework-service.app"), testutil.Contains, `member={RequestName,ReleaseName,GetConnectionCredentials}`)
```

Comment:
It is fine to not assign to classicSlot AFAICS. In fact, you could probably remove it from StorageFrameworkServiceInterfaceSuite since you aren't doing anything with it. If it is decided that storage-framework-service would be provided outside of a snap, then we can start considering classic distro policy.